conmon-rs/server/src/child_reaper.rs

```diff
@@ -0,0 +1,156 @@
+//! Child process reaping and management.
+
+use anyhow::{bail, Result};
+use getset::Getters;
+use log::{debug, error};
+use nix::sys::signal::{kill, Signal};
+use nix::sys::wait::{waitpid, WaitStatus};
+use nix::unistd::Pid;
+use std::path::PathBuf;
+use std::thread;
+use std::{collections::HashMap, fs::File, io::Write, sync::Arc, sync::Mutex};
+
+impl ChildReaper {
+    pub fn init_reaper(&self) -> Result<()> {
+        let grandchildren = Arc::clone(&self.grandchildren);
+        let wait_lock = Arc::clone(&self.wait_lock);
+        thread::spawn(move || {
+            loop {
+                // To prevent an error: "No child processes (os error 10)" when spawning new runtime processes,
+                // we don't call waitpid until the runtime process has finished
+                // Inspired by
+                // https://github.com/kata-containers/kata-containers/blob/828a304883a4/src/agent/src/signal.rs#L27
+                let wait_status = {
+                    let wait_lock = Arc::clone(&wait_lock);
+                    let _lock = wait_lock.lock();
+                    waitpid(Pid::from_raw(-1), None)
+                };
+
+                let grandchildren = Arc::clone(&grandchildren);
```

Comment:
This additional clone can be removed.
```suggestion
```

Reply:
if I don't have it I get 
```
error[E0382]: use of moved value: `grandchildren`
  --> conmon-rs/server/src/child_reaper.rs:23:43
   |
23 | ...                   thread::spawn(move || {
   |                                     ^^^^^^^ value moved into closure here, in previous iteration of loop
24 | ...                       if let Err(e) = Self::forget_grandchild(grandchildren, grandchild_pid, exit_status) {
   |                                                                   ------------- use occurs due to use in closure
   |
   = note: move occurs because `grandchildren` has type `Arc<std::sync::Mutex<HashMap<i32, ReapableChild>>>`, which does not implement the `Copy` trait

```